From 786396b81e485c5410f9c684384e474fdb560620 Mon Sep 17 00:00:00 2001
From: Paul Zander <negril.nx+gentoo@gmail.com>
Date: Mon, 4 Aug 2025 18:32:17 +0200
Subject: [PATCH] dev-libs/protobuf: do not propagate abseil-cpp deps

This should fix unneccessary rebuilds for abseil-cpp via protobuf.

Signed-off-by: Paul Zander <negril.nx+gentoo@gmail.com>

diff --git a/cmake/libprotobuf-lite.cmake b/cmake/libprotobuf-lite.cmake
index 9aa81fb..9596f67 100644
--- a/cmake/libprotobuf-lite.cmake
+++ b/cmake/libprotobuf-lite.cmake
@@ -27,7 +27,7 @@ target_include_directories(libprotobuf-lite PUBLIC
   $<BUILD_INTERFACE:${protobuf_SOURCE_DIR}/src>
   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
 )
-target_link_libraries(libprotobuf-lite PUBLIC ${protobuf_ABSL_USED_TARGETS})
+target_link_libraries(libprotobuf-lite PRIVATE ${protobuf_ABSL_USED_TARGETS})
 protobuf_configure_target(libprotobuf-lite)
 protobuf_configure_unity_target(libprotobuf-lite)
 if(protobuf_BUILD_SHARED_LIBS)
diff --git a/cmake/libprotobuf.cmake b/cmake/libprotobuf.cmake
index 11c09b1..f7ffcc0 100644
--- a/cmake/libprotobuf.cmake
+++ b/cmake/libprotobuf.cmake
@@ -30,7 +30,7 @@ target_include_directories(libprotobuf PUBLIC
   $<BUILD_INTERFACE:${protobuf_SOURCE_DIR}/src>
   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
 )
-target_link_libraries(libprotobuf PUBLIC ${protobuf_ABSL_USED_TARGETS})
+target_link_libraries(libprotobuf PRIVATE ${protobuf_ABSL_USED_TARGETS})
 protobuf_configure_target(libprotobuf)
 if(protobuf_BUILD_SHARED_LIBS)
   target_compile_definitions(libprotobuf
diff --git a/cmake/libprotoc.cmake b/cmake/libprotoc.cmake
index 1fd275d..ee2df78 100644
--- a/cmake/libprotoc.cmake
+++ b/cmake/libprotoc.cmake
@@ -17,7 +17,7 @@ if(protobuf_HAVE_LD_VERSION_SCRIPT)
     LINK_DEPENDS ${protobuf_SOURCE_DIR}/src/libprotoc.map)
 endif()
 target_link_libraries(libprotoc PRIVATE libprotobuf)
-target_link_libraries(libprotoc PUBLIC libupb ${protobuf_ABSL_USED_TARGETS})
+target_link_libraries(libprotoc PRIVATE libupb ${protobuf_ABSL_USED_TARGETS})
 protobuf_configure_target(libprotoc)
 if(protobuf_BUILD_SHARED_LIBS)
   target_compile_definitions(libprotoc
diff --git a/cmake/protoc.cmake b/cmake/protoc.cmake
index 385a7a3..d37d98a 100644
--- a/cmake/protoc.cmake
+++ b/cmake/protoc.cmake
@@ -4,6 +4,7 @@ set(protoc_files
 
 add_executable(protoc ${protoc_files} ${protobuf_version_rc_file})
 target_link_libraries(protoc
+  PRIVATE
   libprotoc
   libprotobuf
   ${protobuf_ABSL_USED_TARGETS}
-- 
2.50.1

